acceptance/suites/tests/authorization/hup_reloads_crl.rb

```diff
@@ -0,0 +1,62 @@
+test_name '(SERVER-1380) CRL should reload after a server HUP' do
+
+server = master.puppet['certname']
+
+teardown do
+  # Leave agents with working certs. Destroy the revoked cert on the master,
+  # remove all traces on the agent, make the agent request a shiny new cert, 
+  # then have the master sign it.
+  agents.each do |a|
+    if not_controller(a)
+      nodefqdn = fact_on(a, 'fqdn')
+      ssldir = a.puppet['ssldir']
+      reqdir = a.puppet['requestdir']
+      on(master, puppet('cert', 'destroy', "#{nodefqdn}"))
+
+      # FIXME: this is a dirty nasty hack that Beaker should
+      # handle seamlessly
+      hostcert_file = a.puppet['hostcert']
+      request_file = File.join(reqdir, nodefqdn + ".pem")
+      on(a, "rm -v #{hostcert_file} #{request_file}",
+         {:accept_all_exit_codes => true})
+      on(a, puppet('agent', '--test', '--server', "#{server}"),
+         {:acceptable_exit_codes => [1]})
+      on(master, puppet('cert', 'sign', "#{nodefqdn}"),
+         {:accept_all_exit_codes => true})
+    end
+  end
+end
+
+step 'Demonstrate that the existing certs work' do
+  agents.each do |a| 
+    if not_controller(a)
+      rc = on(a, puppet('agent', '--test', '--server', "#{server}"),
+              {:acceptable_exit_codes => [0,2]})
+    end
+  end
+end
+
+step 'Revoke the agent certs' do
+  agents.each do |a| 
+    if not_controller(a)
```

Comment:
My previous objection above doesn't really apply, because there are no cases in our CI pipelines that I can find where we are running with a master only test environment.